### ISSUE TITLE
Add a entity class Technique

### ISSUE THREAD

**Issue by RKrahl** [icatproject/icat.server#230]
The [current draft for a data model](https://github.com/panosc-eu/search-api/blob/master/doc/data-model.md) for the PaNOSC WP3 search API envisages an entity class `Technique`. We currently do not have that in our schema. As discussed in the last F2F meeting, we might want to adopt this idea.

A `Technique` should have the attributes `name` and `pid` and have a many-to-many relation with `Dataset`. I therefore suggest to add the following new classes to the schema:

## `Technique`

Represents an experimental technique

Constraint: `facility`, `name`

Relationships:

Card | Class            | Field
-----|------------------|---------------------
1,1  | Facility         | facility
0,*  | DatasetTechnique | datasetTechniques

Other fields:

Field       | Type                 | Description
------------|----------------------|-------------------------------
name        | String[255] NOT NULL | A short name for the technique
pid         | String[255]          | A persistent identifier
description | String[255]          | An informal description

## `DatasetTechnique`

Represents a many-to-many relationship between a dataset and the experimental technique being used to create that Dataset

Constraint: `dataset`, `technique`

Relationships:

Card | Class     | Field
-----|-----------|----------
1,1  | Dataset   | dataset
1,1  | Technique | technique

**Comment by agbeltran** [comment]
For the description of ```pid``` in the ```Technique``` table, I would add that it should be ideally pointing to a vocabulary term, thus the text would look like  ```"A persistent identifier attributed to the technique, ideally referring to a vocabulary term". ```

**Comment by RKrahl** [comment]
It has been suggested in the monthly meeting to remove the relation to `Facility`. I'm happy with the suggestion. I'd just like to add that the purpose of the `Facility` table in general might need some reflection, but this would then be another issue.

**Comment by dfq16044** [comment]
Rolf, for the technique pid do you mean a local (within the facility) pid? 
I know that we can have PID for instruments, samples ...
If not a local pid, are you thinking of using a general PID provider like https://www.pidconsortium.net/?

**Comment by RKrahl** [comment]
The `pid` attribute of `Technique` is intended to be used as described by @agbeltran: if we (for instance in ExPaNDS) come up with an ontology or at least with a controlled vocabulary covering experimental techniques, the terms in that vocabulary may each be identified with a persistent identifier. This PID should then go into the `pid` attribute. (And yes, I agree with your suggestion for the improved description of that attribute, Alejandra.)

Which PID provider we are going to use for the vocabulary is not yet decided and the proposed ICAT schema extension is agnostic towards that question. Note furthermore that the `pid` attribute is optional. It can be left undefined as long as we do not have a vocabulary with terms identified by PIDs.

**Comment by agbeltran** [comment]
I agree with the changed we discussed in the last meeting about removing ```Facility``` (raised by @kevinphippsstfc). However, I think it might be useful to have a relationship between ```Technique``` and ```Instrument``` so that we can know what instruments implement a particular technique. We can discuss this in a separate issue.

 Just to summarise the discussion, the new version of ```Technique``` would look like:

## `Technique`

Represents an experimental technique

Constraint: `facility`, `name`

Relationships:

Card | Class            | Field
-----|------------------|---------------------
0,*  | DatasetTechnique | datasetTechniques

Other fields:

Field       | Type                 | Description
------------|----------------------|----------------------------------------------------
name        | String[255] NOT NULL | A short name for the technique
pid         | String[255]          | A persistent identifier attributed to the technique, ideally referring to a vocabulary term.
description | String[255]          | An informal description for the technique.

## `DatasetTechnique`

Represents a many-to-many relationship between a dataset and the experimental technique being used to create that Dataset

Constraint: `dataset`, `technique`

Relationships:

Card | Class     | Field
-----|-----------|----------
1,1  | Dataset   | dataset
1,1  | Technique | technique

## Affects to existing classes

As an implicit result of adding `DatasetTechnique`, a new relationship would be added to `Dataset`:

Card | Class            | Field
-----|------------------|---------------------
0,*  | DatasetTechnique | datasetTechniques

**Comment by dfq16044** [comment]
The problem of associated Instrument with Technique is that users can bring their own detector the facility ( a detector will be associated to a particular technique in general). 
This means that in this use case, the detector is not attached to the Instrument permanently. 
I will let you create an issue to discuss that topic

**Comment by agbeltran** [comment]
Thanks @dfq16044 - given your comment, then probably the ```Instrument``` \ ```Technique``` relationship should not be added, and it would be enough to associate ```Dataset``` and ```Technique```.

In fact, revising the [PANOSC API model](https://github.com/panosc-eu/search-api/tree/master/common/models), indeed techniques are associated with datasets and not instruments.

**Comment by agbeltran** [comment]
Related to this addition to the ICAT schema, since the previous discussions the initial version of the PaNET ontology for Photon and Neutro Experimental Techniques is now available, and it was a deliverable from ExPaNDs. 

The ontology development is here:
https://github.com/ExPaNDS-eu/ExPaNDS-experimental-techniques-ontology

The ontology documentation is here:
https://expands-eu.github.io/ExPaNDS-experimental-techniques-ontology/index-en.html

The ontology can be visualised in BioPortal:
https://bioportal.bioontology.org/ontologies/PANET

The related ExPANDS deliverable is here:
https://doi.org/10.5281/zenodo.4806026

**Comment by agbeltran** [comment]
One point for discussion is the label ```pid``` - perhaps we can use the more general label ```identifier```? As long as we harmonise the label across all the entities.

**Comment by agbeltran** [comment]
The ICAT collaboration met on 10th August and the agreement was to:
- implement the ```Technique``` class
- implement the ```DatasetTechnique``` class
as per proposal at https://github.com/icatproject/icat.server/issues/230#issuecomment-652959419

We will discuss ```InstrumentTechnique``` in a separate issue.
